Source/Task/AsyncQueue.cpp

```diff
@@ -351,9 +351,12 @@ class Queue
     bool Wait(uint32_t timeout)
     {
         std::unique_lock<std::mutex> lock(m_cs);
-        if (!m_hasItems)
+        while (!m_hasItems)
```

Comment:
there's a version of wait_for that takes a lambda with the predicate, saves you writing the loop and the break
```
m_event.wait_for(lock, std::chrono::milliseconds(timeout), [=](){ return !m_hasItems; });
```